pkg/services/ngalert/notifier/channels/discord.go

```diff
@@ -282,21 +270,32 @@ func (d DiscordNotifier) buildRequest(ctx context.Context, url string, body []by
 	if err != nil {
 		return nil, err
 	}
+
 	if _, err := payload.Write(body); err != nil {
 		return nil, err
 	}
+
 	for _, a := range attachments {
-		part, err := w.CreateFormFile("", a.name)
-		if err != nil {
-			return nil, err
-		}
-		if _, err := io.Copy(part, a.reader); err != nil {
-			return nil, err
+		if a.reader != nil { // We have an image to upload.
```

Comment:
I think this will panic without the nil check. io.Copy doesn't check for nillity.

Reply:
I'll write a test for that, of course.